### PR TITLE
Code Refactoring for client tests

### PR DESCRIPTION
**Pull request by windkit**
#### Unified Test Items
- PUT Object
- Multipart Upload Object
- HEAD Object
- GET Object
- Range GET Object
- COPY Object
- List Object
- DELETE Object
- Paged List Object
- Multi-DELETE Object
- GET Bucket ACL
- SET Bucket ACL

#### Supported SDKs / Libraries
- aws-sdk-java
- aws-sdk-php
- aws-sdk-go
- aws-sdk-ruby
- boto (Python)

#### WIP SDKs / Libraries
- erlcloud (Erlang)
- jcloud (Java)
- s3cmd (Bash)

#### Code Refactoring
Global Options
- Host / Port
- Access / Secret Key
- Bucket
- Temp Data Path (testFile and testFile.large)
- Sign Version (aws-sdk-go only supports V4)

### CODE REVIEW

File: aws-sdk-go/LeoFSTest.go [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+func putDummyObjects(bucketName, prefix string, total int, holder string) {
+    bucketStr := aws.String(bucketName)
+    for i := 0; i < total; i++ {
+        reader, _ := os.Open(holder)
+        svc.PutObject(&s3.PutObjectInput{
+            Bucket  : bucketStr,
+            Key     : aws.String(fmt.Sprintf("%s%d", prefix, i)),
+            Body    : reader,
+        })
+        reader.Close()
```
Comment (mocchira) [inline aws-sdk-go/LeoFSTest.go]: Change `defer reader.Close()` and place it at the next line of `reader, _ := os.Open(holder)`

File: aws-sdk-go/LeoFSTest.go [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+    }
+}
+
+//Ref: http://play.golang.org/p/S2s3e0nj4s
+func doesFileMatch(r1, r2 io.Reader) bool {
+    buf1 := bufio.NewReader(r1)
+    buf2 := bufio.NewReader(r2)
+    for {
+        b1, err1 := buf1.ReadByte()
+        b2, err2 := buf2.ReadByte()
```
Comment (mocchira) [inline aws-sdk-go/LeoFSTest.go]: Using `Read(p []byte) (n int, err error)` and compare multiple bytes at once by using http://golang.org/pkg/bytes/#Compare seems to be more efficient.

File: aws-sdk-java/LeoFSTest.java [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+    private static final String smallTestF  = tempData + "testFile";
+    private static final String largeTestF  = tempData + "testFile.large"; 
+
+    private static ClientConfiguration config;
+    private static AWSCredentials credentials;
+    private static AmazonS3 s3;
+
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        // Init
```
Comment (mocchira) [inline aws-sdk-java/LeoFSTest.java]: As we keep doing tests with authentication v2 for a while, 
Please make signVer configurable via a command line argument (not only java but all s3 clients except one which don't have v2 signer like aws-sdk-go.

**Comment by mocchira** [comment]
@windkit LGTM.
As your base is a bit outdated, 
can you rebase?

**Comment by windkit** [comment]
Rebased, I guess I will open another PR for erlcloud and jclouds afterwards.

**Comment by mocchira** [comment]
@windkit 

> Rebased, I guess I will open another PR for erlcloud and jclouds afterwards.

Got it. Thanks.

**Pull request merged**